src/main/java/org/graylog/plugins/threatintel/functions/otx/OTXIPLookupFunction.java

```diff
@@ -23,40 +19,26 @@
 
     private final ParameterDescriptor<String, String> valueParam = ParameterDescriptor
             .string(VALUE)
-            .description("The IPv4 or IPv6 address to look up. Example: 198.51.100.1 or 2001:0db8:85a3:0000:0000:8a2e:0370:7334"
-            ).build();
+            .description("The IPv4 or IPv6 address to look up. Example: 198.51.100.1 or 2001:0db8:85a3:0000:0000:8a2e:0370:7334")
+            .build();
 
-    private final OTXLookupProvider provider;
 
     @Inject
-    public OTXIPLookupFunction(final ClusterConfigService clusterConfigService,
-                               final MetricRegistry metricRegistry) {
-        OTXIPLookupProvider.getInstance().initialize(clusterConfigService, metricRegistry);
-
-        this.provider = OTXIPLookupProvider.getInstance();
-    }
-
-    @Override
-    public Object preComputeConstantArgument(FunctionArgs args, String s, Expression arg) {
-        return arg.evaluateUnsafe(EvaluationContext.emptyContext());
+    public OTXIPLookupFunction(final LookupTableService lookupTableService) {
+        super(lookupTableService);
     }
 
     @Override
     public OTXLookupResult evaluate(FunctionArgs args, EvaluationContext context) {
-        String ip = valueParam.required(args, context);
+        final String ip = valueParam.required(args, context);
         if (ip == null) {
             LOG.error("NULL parameter passed to OTX threat intel lookup.");
             return null;
```

Comment:
Same here, maybe `OTXLookupResult.EMPTY` would be better than `null`?

Reply:
✅